CMakePresets.json

```diff
@@ -57,6 +78,18 @@
         "strategy": "external"
       }
     },
+    {
+      "name": "x86-msvc-static",
+      "displayName": "Windows x86 MSVC Static",
+      "description": "Windows Default, MSVC, x86 architecture.",
+      "inherits": "msvc-windows-default",
+      "hidden": true,
+      "cacheVariables": {
+        "VCPKG_TARGET_TRIPLET": "x86-windows-static",
+        "MSVC_USE_STATIC_CRT": true
+      },
+      "architecture": "win32"
```

Comment:
Should we add this `architecture` property to the other x86-* presets being added below?

Reply:
All the x86 presets being added should derive from the MSVC base and the way cmake presets work, derived presets include all the settings from their parent.

Reply:
It's part of why I love the cmakepresets.json stuff - it would be nice if there was platform matrix support like we have in the CI pipelines, but it's really nice that you get a hierarchy of types.